src/prediction_result_formatter_bed_file.py

```diff
@@ -0,0 +1,41 @@
+from typing import Iterable
+import numpy as np
+import time
+from src.prediction_result_formatter import PredictionResultFormatter
+from src.prediction_result import PredictionResult
+
+class PredictionResultFormatterBedFile(PredictionResultFormatter):
+    def file_name(self, prediction_result: PredictionResult) -> str:
+        model_params_as_string_for_file_name = prediction_result.get_model_params_as_string_for_file_name()
+        now_time_as_string_for_file_name = time.strftime("%Y_%m_%d,%H_%M_%S")
+        sequence_variation = prediction_result.sequence_variation
+        file_name = prediction_result.file_name
+        seq_record_name = prediction_result.seq_record_name
+        seq_record_key = '{}.{}.{}.{}'.format(file_name, seq_record_name, model_params_as_string_for_file_name, now_time_as_string_for_file_name)
+        seq_name = sequence_variation.get_title()
+        seq_key = '{}.{}'.format(seq_record_key, seq_name)
+        bed_file_name_seq = '{}.bed'.format(seq_key)
+        #bed_file_name = '{}.bed'.format(seq_record_key)
+
+        return bed_file_name_seq
+        
+    def format(self, prediction_result: PredictionResult) -> Iterable[str]:
+        sequence_variation = prediction_result.sequence_variation
+        model_params_as_string = prediction_result.get_model_params_as_string()
+        seq_name = sequence_variation.get_title()
+        minimum_sequence_length = prediction_result.minimum_sequence_length
+        seq_len = prediction_result.seq_len
+        max_label = prediction_result.max_label
+        labeled = prediction_result.labeled
+        
+        label_id = 1
+        for label in range(1, max_label+1):
+            candidate = np.where(labeled == label)[0]
+            candidate_length = candidate.shape[0]
+            if candidate_length>minimum_sequence_length:
+                bed_name = '{},{},{}'.format(model_params_as_string, seq_name, label_id)
+                
+                candidate_start, candidate_end = sequence_variation.derive_candidate_start_and_end(seq_len, candidate)
+                yield '0\t{start}\t{end}\t{name}'.format(start=candidate_start, end=candidate_end, name=bed_name)
```

Comment:
a requested feature of yours was to be able to specify the chromosome name, is that correct @AlistairNGarratt ?

```suggestion
                yield '{chromosome_name}\t{start}\t{end}\t{name}'.format(start=candidate_start, end=candidate_end, name=bed_name, chromosome_name=chromosome_name)
```

Reply:
But it doesn't make complete sense in some contexts where it's possible to process multiple files consecutively. Although having an option is always in option.

So let's move this to a new issue if need be.

Reply:
I think it would be great if the .bed files included a name, such as chr1, that can be specified by the user, in the first column. Note that there should be no quotation marks in the name, otherwise it is not compatible with other .bed tools. When processing multiple files simultaneously, e.g. multiple chromosomes, it would be an advantage to have this chromosome designation in the first column in order to distinguish the .bed file entries later when merging within a spreadsheet.

Reply:
I created a new issue for this, because it's a feature request: https://github.com/AnthyG/Z-DNABERT/issues/9#issue-2748854933